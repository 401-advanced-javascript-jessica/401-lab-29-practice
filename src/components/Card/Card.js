import React from 'react';
import PropTypes from 'prop-types';
import { If } from '../if/index';

const Card = (props) => {
  const { card } = props;
  return (
        <div>
            <div className="card">
                <If condition={card.title}>
                    <header><h3>{card.title}</h3></header>
                </If>

                <div className="content">{card.copy}</div>
                <figure>
                    <img src={card.media.href} alt={card.media.alt} />
                    <figcaption>{card.media.title}</figcaption>
                </figure>
                <nav className="links">
                    <ul>
                        {card.links.map((link, i) => (
                            <li key={i}>
                                <a className={link.type}
                                   href={link.href}
                                   title={link.title}>
                                    {link.title}
                                </a>
                            </li>
                        ))}
                    </ul>
                </nav>
            </div>
        </div>
  );
};

Card.propTypes = {
  card: PropTypes.object,
};

export default Card;
