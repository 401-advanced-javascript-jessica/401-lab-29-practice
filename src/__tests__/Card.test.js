import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../components/Card/Card';


describe('<Card />', () => {
  const card = {
    title: 'Content Record 1',
    copy: 'Content Record 1 Copy Here...',
    media: {
      alt: 'This is a picture',
      title: 'This is the title of the picture',
      href: 'https://placehold.it/200x100',
    },
    links: [
      {
        type: 'primary',
        title: 'Click Me',
        href: 'https://www.foo.com',
      },
      {
        type: 'secodary',
        title: 'Not Me',
        href: 'https://www.bar.com',
      }],
  };

  it('Card is rendered at application start', () => {
    const app = shallow(<Card card={ card }/>);
    expect(app.find('.card').exists()).toBe(true);
  });

  it('content is rendered', () => {
    const app = mount(<Card card={ card }/>);
    expect(app.find('h3').text()).toEqual('Content Record 1');
  });

  it('renders properly', () => {
    const tree = renderer.create(<Card card={ card }/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
