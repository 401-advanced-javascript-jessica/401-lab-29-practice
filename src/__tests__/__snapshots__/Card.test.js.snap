// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders properly 1`] = `
<div>
  <div
    className="card"
  >
    <header
      condition="Content Record 1"
    >
      <h3>
        Content Record 1
      </h3>
    </header>
    <div
      className="content"
    >
      Content Record 1 Copy Here...
    </div>
    <figure>
      <img
        alt="This is a picture"
        src="https://placehold.it/200x100"
      />
      <figcaption>
        This is the title of the picture
      </figcaption>
    </figure>
    <nav
      className="links"
    >
      <ul>
        <li>
          <a
            className="primary"
            href="https://www.foo.com"
            title="Click Me"
          >
            Click Me
          </a>
        </li>
        <li>
          <a
            className="secodary"
            href="https://www.bar.com"
            title="Not Me"
          >
            Not Me
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
